bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/ScriptTransformationService.java

```diff
@@ -210,8 +214,16 @@ public void deactivate() {
                     scriptRecord.compiledScript = compiledScript;
                 }
 
-                Object result = compiledScript != null ? compiledScript.eval() : engine.eval(scriptRecord.script);
-                return result == null ? null : result.toString();
+                try {
+                    Object result = compiledScript != null ? compiledScript.eval() : engine.eval(scriptRecord.script);
+                    return result == null ? null : result.toString();
+                } finally {
+                    if (injectedParams != null) {
+                        for (String param : injectedParams) {
```

Comment:
I would prefer `forEach`, but this is just a suggestion.

```suggestion
                                                injectedParams
                                .forEach(param -> executionContext.removeAttribute(param, ScriptContext.ENGINE_SCOPE));
```

Reply:
good idea, that looks nicer indeed